src/t_zset.c

```diff
@@ -2694,7 +2696,9 @@ void zunionInterDiffGenericCommand(client *c, robj *dstkey, int numkeysIndex, in
                 }
 
                 /* Only continue when present in every input. */
-                if (j == setnum) {
+                if (j == setnum && cardinality_only) {
+                    cardinality++;
+                } else if (j == setnum) {
```

Comment:
Nitpick, but maybe this

```suggestion
                if (j == setnum) {
                    if (cardinality_only) {
                        cardinality++;
                    } else {
```

Reply:
It was like that, I changed it to avoid the indentation.